back-end/helpers/user-helpers.js

```diff
@@ -73,6 +74,37 @@ async function getFullUserProfile(userID) {
   return userData;
 }
 
+async function getFullConnectionProfile(requestedID, userID) {
+  const query = 'SELECT users.* FROM users ' +
+    'INNER JOIN connections ON ' +
+    'users.id = connections.user1_id OR users.id = connections.user2_id ' +
+    'WHERE users.id = $1 ' +
+    'AND ((connections.user1_id = $1 AND connections.user2_id = $2) ' +
+    'OR (connections.user1_id = $2 AND connections.user2_id = $1))' +
+    'AND connections.connection_accepted = true'
+  ;
+
+  const connectionData = await queryWithParameters(query, [requestedID, userID]);
+
+  if (!connectionData) {
+    logger.warn(`Unable to find user. Ensure that a valid connection is made. User ID: ${userID}, Requested ID: ${requestedID}`)
+    return null;
+  }
+
+  logger.info(`Retrieved data for id ${requestedID}`)
+
+  delete connectionData.password;
+  const connectionGithubData = await getUserGitHubInfo(requestedID);
+  if (!connectionGithubData) {
```

Comment:
I believe if a person doesn't have their GitHub profile attached, they won't be able to make connections anyway?

Reply:
You're probably right, but I figured it'd just be good to log in case something unforeseen and wonky happens. Thoughts?

Reply:
Yeah it's a good to have just in case. I'm going to need to reflect that on the frontend as well 👍